internal/collections/map.go

```diff
@@ -116,31 +120,24 @@ func (m Map[K, V]) Delete(k K) {
 	delete(m.elems, uniq)
 }
 
-// Elems exposes the internal underlying representation of the map directly,
-// as a pragmatic compromise for efficient iteration.
-//
-// The result of this function is part of the internal state of the receiver
-// and so callers MUST NOT modify it. If a caller is using locks to ensure
-// safe concurrent access then any reads of the resulting map must be
-// guarded by the same lock as would be used for other methods that read
-// data from the reciever.
+// All returns an iterator over the elements of the map, in an unspecified
+// order.
 //
-// The only correct use of this function is as part of a "for ... range"
-// statement using only the values of the resulting map:
+// This is intended for use in a range-over-func statement, like this:
 //
-//	for _, elem := range map.Elems() {
-//	    k := elem.K
-//	    v := elem.V
-//	    // ...
+//	for k, v := range map.All() {
+//		// do something with k and/or v
 //	}
 //
-// Do not access or make any assumptions about the keys of the resulting
-// map. Their exact values are an implementation detail of the receiver.
-func (m Map[K, V]) Elems() map[UniqueKey[K]]MapElem[K, V] {
-	// This is regrettable but the only viable way to support efficient
-	// iteration over map elements until Go gains support for range
-	// loops over custom iterator functions.
-	return m.elems
+// Modifying the map during iteration causes unspecified results. Modifying
+// the map concurrently with advancing the iterator causes undefined behavior
+// including possible memory unsafety.
+func (m Map[K, V]) All() iter.Seq2[K, V] {
+	return func(yield func(K, V) bool) {
+		for _, elem := range m.elems {
+			yield(elem.K, elem.V)
```

Comment:
To break the loop, it should be rewritten as follows. Otherwise, it may panic.

```suggestion
			if !yield(elem.K, elem.) {
				return
			}
```

Reply:
Thanks, that's a good point! We don't currently have any callers that break out of iteration early, but the implementation should definitely permit that.